Fix incremental rollouts backwards compat upon app updates

aitorvs opened the pull request:
Task/Issue URL: https://app.asana.com/0/488551667048375/1205213514616979/f

### Description
Incremental rollouts did not maintain backwards compatibility upon app update

Bug repro steps:
- [ ] build and install play debug build from 4e1221a99 (before incremental rollouts)
- [ ] verify "Autofill logins" appears under app settings
- [ ] build and update play build from 11b2c9abc (incremental rollouts)
- Expected:
  - [ ] "Autofill logins" are present under app settings
- Actual
  - [ ] "Autofill logins" are NOT present under app settings

### Steps to test this PR
Bug repro steps:
- [x] build and install play debug build from 4e1221a99 (before incremental rollouts)
- [x] verify "Autofill logins" appears under app settings
- [x] build and update play build from 11b2c9abc (incremental rollouts)
- [x] verify "Autofill logins" are no longer present under app settings (BUG)
- [ ] build and update play build this branch
- [ ] Verify "Autofill logins" are present under app settings

aitorvs commented:
Current dependencies on/for this PR:
* develop
  * **PR #3419** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3419" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3419?utm_source=stack-comment).

CDRussell commented on feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt:
```diff
@@ -108,14 +110,35 @@ class FeatureToggles private constructor(
 
+    /**
+     * The usage of this API is only useful for internal/dev settings/features
+     * If you find yourself having to call this method in production code, then YOUR DOING SOMETHING WRONG
```
typo: `your` should be `you're`

aitorvs replied:
Done

CDRussell commented on feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt:
```diff
@@ -108,14 +110,35 @@ class FeatureToggles private constructor(
 
+    /**
+     * The usage of this API is only useful for internal/dev settings/features
+     * If you find yourself having to call this method in production code, then YOUR DOING SOMETHING WRONG
```
typo: `your` should be `you're`

aitorvs replied:
done

CDRussell commented on feature-toggles/feature-toggles-api/src/main/java/com/duckduckgo/feature/toggles/api/FeatureToggles.kt:
```diff
@@ -108,14 +110,35 @@ class FeatureToggles private constructor(
+     * The usage of this API is only useful for internal/dev settings/features
+     * If you find yourself having to call this method in production code, then YOUR DOING SOMETHING WRONG
+     *
+     * @return the raw [State] store for this feature flag.
```
Can you add a line explaining what the raw state means? I understand that `isEnabled` is the important API here generally, but i'm not sure what raw state really is from reading this.

aitorvs replied:
Done

CDRussell commented on feature-toggles/feature-toggles-impl/src/test/java/com/duckduckgo/feature/toggles/api/FeatureTogglesTest.kt:
```diff
@@ -27,12 +30,14 @@ class FeatureTogglesTest {
 
```
Not a change in this PR but while you're here... can you double-check `whenEnabledByDefaultAndSetDisabledThenReturnDisabled()`... not sure, but wonder if it should be using `feature.enabledByDefault()` on both lines instead of different features on each?

aitorvs replied:
oh, good catch. Fixed

CDRussell reviewed (changes requested):
This does not work for a fresh install. A fresh install on this branch results in autofill disappearing again.



(upgrade path works ok as you've described in the testing steps).

aitorvs commented:
> This does not work for a fresh install. A fresh install on this branch results in autofill disappearing again.
> 
> (upgrade path works ok as you've described in the testing steps).

Addressed the comments and fixed (and added specific test) for the fresh install and subsequent update

CDRussell reviewed (approved):
Re-tested and can confirm: Fresh install path now has autofill enabled.
